policy-controller/k8s/index/src/index.rs

```diff
@@ -110,6 +112,7 @@ struct PolicyIndex {
     server_authorizations: HashMap<String, server_authorization::ServerAuthz>,
 
     authorization_policies: HashMap<String, authorization_policy::Spec>,
+    http_routes: HashMap<String, RouteBinding>,
```

Comment:
Note that spec indicates that we need to be able to prefer older HttpRoutes to new ones when a match is otherwise equivalent. In the Proxy API, we want this to be expressed by ordering. I expect that we either want to avoid `HashMap` -- so that the routes are stored in a predictable order -- or otherwise maintain information that allows us to sort the results before returning via the API.